opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -87,23 +87,50 @@ message InstrumentationLibraryMetrics {
 //                       +---------------------------+
 //
 //-----------------------------------------------------------------------
-// DataPoint is a value of specific type corresponding to a given moment in
-// time. Each DataPoint is timestamped.
-//
-// DataPoint is strongly typed: each DataPoint type has a specific Protobuf message
-// depending on the value type of the metric and thus there are currently 4 DataPoint
-// messages, which correspond to the types of metric values.
 message Metric {
   // metric_descriptor describes the Metric.
   MetricDescriptor metric_descriptor = 1;
 
-  // Data is a list of one or more DataPoints for a single metric. Only one of the
-  // following fields is used for the data, depending on the type of the metric defined
-  // by MetricDescriptor.type field.
-  repeated Int64DataPoint int64_data_points = 2;
-  repeated DoubleDataPoint double_data_points = 3;
-  repeated HistogramDataPoint histogram_data_points = 4;
-  repeated SummaryDataPoint summary_data_points = 5;
+  repeated Data data = 2;
+
+  message Data {
+    // The set of labels that uniquely identify this timeseries.
+    //
+    // These labels take precedence over any labels with the same key defined
+    // in the metric descriptor.
+    repeated opentelemetry.proto.common.v1.StringKeyValue labels = 1;
+
+    // start_time_unix_nano is the beginning of a time interval over which
+    // this data point was recorded. It MUST be represented as the UNIX Epoch
+    // time in nanoseconds (nanoseconds since 00:00:00 UTC on 1 January 1970).
+    // If zero, it is treated as unspecified and may be set by the receiver.
+    //
+    // If this data point is a part of a timeseries for an INSTANTANEOUS
+    // metric, this value is ignored.
+    //
+    // If this data point is a part of a timeseries for an DELTA, or
+    // CUMULATIVE metric, this value represents the instant before the
+    // measuring time interval began (i.e. for time t to be in the measuring
+    // interval than start_time_unix_nano < t <= time_unix_nano).
+    //
+    // This time MUST be reset so as to represent a new measurement lifetime
+    // if this data point is part of a timeseries for a CUMULATIVE metric and
+    // the metric value is reset to zero.
+    fixed64 start_time_unix_nano = 2;
+
+    // time_unix_nano is the time when this data point was recorded. It MUST
+    // be represented as the UNIX Epoch time in nanoseconds (nanoseconds since
+    // 00:00:00 UTC on 1 January 1970). If zero, it is treated as unspecified
+    // and may be set by the receiver.
+    fixed64 time_unix_nano = 3;
+
+    // Value is the measured value.
+    oneof Value {
```

Comment:
This means `metric.data` might have a mix of value types right? Vs. https://github.com/open-telemetry/opentelemetry-proto/pull/137/files#diff-ad849336b0a70e8a18346421908f69edL103-L106.

Reply:
I'm debating whether the descriptor should indicate the expected Data field, or the Data struct should do so itself. I can see it both ways. It's simpler if the descriptor identifies a constant field to be expected. I've also described how I believe that we should be able to report single-value distributions using a single value, which means that an observer reporting a distribution should report a scalar value when there's no aggregation, but should report a distribution value when there is aggregation.

Reply:
Yeah, it currently does. I think (for a variety of reasons) I'm going to go back to a format like before.

I think it is still possible in the old format for the sender to send a mix of data, but with the distinct types it is likely easier to ignore this error.

Reply:
@jmacd : That's a good point. I'm not sure either. I can see the inclusion of the type in the `Data` itself allowing for simpler extension of a metric instrument (likely going to be really useful when there are multiple views on the same instrument), but also how it might be easier to reason about distinct metric descriptor that fully defines the data.

Reply:
I pushed a change to put this info in the descriptor. If you come to a different conclusion @jmacd, let me know.